src/borg/archiver.py

```diff
@@ -1145,12 +1145,16 @@ def _delete_archives(self, args, repository):
         stats = Statistics()
         with Cache(repository, key, manifest, progress=args.progress, lock_wait=self.lock_wait) as cache:
             for i, archive_name in enumerate(archive_names, 1):
-                msg = 'Would delete archive: {} ({}/{})' if dry_run else 'Deleting archive: {} ({}/{})'
-                logger.info(msg.format(format_archive(manifest.archives[archive_name]), i, len(archive_names)))
-                if not dry_run:
-                    archive = Archive(repository, key, manifest, archive_name, cache=cache,
-                                      consider_part_files=args.consider_part_files)
-                    archive.delete(stats, progress=args.progress, forced=args.forced)
+                try:
+                    msg = 'Would delete archive: {} ({}/{})' if dry_run else 'Deleting archive: {} ({}/{})'
+                    logger.info(msg.format(format_archive(manifest.archives[archive_name]), i, len(archive_names)))
+                    if not dry_run:
+                        archive = Archive(repository, key, manifest, archive_name, cache=cache,
```

Comment:
if we want similar behavior as do_list for a unknown archive, we could create the Archive instance first, like:
```
arch = Archive(...)  # if it does not exist, that will raise an Exception leading to a clean error message (as in do_list)
msg = ...
logger.info(...)
if not dry_run:
    arch.delete(...)

Reply:
True it is.
Sorry for being away this week, I was sick and away from almost everything :/
Patch coming soon.